syntax = "proto3";
package huzch;
import "base.proto";

option cc_generic_services = true;

message GetHistoryMessageReq {
    string request_id = 1;
    string chat_session_id = 2;
    int64 start_time = 3;
    int64 end_time = 4;
    optional string user_id = 5;
    optional string login_session_id = 6;
}
message GetHistoryMessageRsp {
    string request_id = 1;
    bool success = 2;
    optional string errmsg = 3; 
    repeated MessageInfo messages_info = 4;
}

message GetRecentMessageReq {
    string request_id = 1;
    string chat_session_id = 2;
    int64 msg_count = 3;
    optional int64 cur_time = 4; // 用于扩展获取指定时间前的n条消息
    optional string user_id = 5;
    optional string login_session_id = 6;
}
message GetRecentMessageRsp {
    string request_id = 1;
    bool success = 2;
    optional string errmsg = 3; 
    repeated MessageInfo messages_info = 4;
}

message MessageSearchReq {
    string request_id = 1;
    string search_key = 2;
    string chat_session_id = 3;
    optional string user_id = 4;
    optional string login_session_id = 5;
}
message MessageSearchRsp {
    string request_id = 1;
    bool success = 2;
    optional string errmsg = 3; 
    repeated MessageInfo messages_info = 4;
}

service MessageService {
    rpc GetHistoryMessage(GetHistoryMessageReq) returns (GetHistoryMessageRsp);
    rpc GetRecentMessage(GetRecentMessageReq) returns (GetRecentMessageRsp);
    rpc MessageSearch(MessageSearchReq) returns (MessageSearchRsp);
}